### PR TITLE
fix(queue): handle queue config API code 500 when config file is invalid

### PR DESCRIPTION
**Pull request by Syffe**
queue configuration API wasn't handling the case where a config file requested is invalid,
thus returning 500 INTERNAL SERVER ERROR. We know handle the case and return a proper payload.

Fixes MRGFY-1200

### CODE REVIEW

File: mergify_engine/web/api/queues.py [hunk 1]
```diff
@@ -344,7 +345,13 @@ async def repository_queues_configuration(
     if config_file is None:
         return QueuesConfig()
 
-    config = get_mergify_config(config_file)
+    try:
+        config = get_mergify_config(config_file)
+    except InvalidRules:
+        raise fastapi.HTTPException(
+            status_code=409,
+            detail="The configuration file rules are invalid. Please verify they are valid or up to date.",
```
Comment (jd) [inline mergify_engine/web/api/queues.py:353]:
```suggestion
            detail="The configuration file is invalid.",
```

keep it simple.

File: mergify_engine/web/api/queues.py [hunk 2]
```diff
@@ -344,7 +345,13 @@ async def repository_queues_configuration(
     if config_file is None:
         return QueuesConfig()
 
-    config = get_mergify_config(config_file)
+    try:
+        config = get_mergify_config(config_file)
+    except InvalidRules:
+        raise fastapi.HTTPException(
+            status_code=409,
```
Comment (jd) [inline mergify_engine/web/api/queues.py:352]: You need to add this return type in the documentation (see decorator).

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/web/api/queues.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/web/api/queues.py:352]: Shouldn't be 422?

Comment (Syffe) [reply mergify_engine/web/api/queues.py:352]: Indeed 409 is more for a conflict in the request, in this case the request is good, only the ressource/entity has conflict

**Review by jd — APPROVED** [review]

**Pull request merged**